### PR TITLE
JAPI-131: Initial check-in for common error classes

### PR DESCRIPTION
**Pull request by drealeed**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [X] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

To give you a leg up, I've created a new hpcc-commons project with the two new error classes for use in clienttools, wsclient, etc.: HpccError and HpccErrorBlock.

I named the project hpcc-commons to distinguish it from the existing "commons" project we have for hpccsystems.commons.

I based HpccErrorBlock off of the ErrorBlock class and HpccError off of the HError class, giving them different names to avoid confusion and full package naming in classes where both are used.

You'll notice that the HpccErrorCode, HpccErrorType and HpccErrorSource enums implement a corresponding interface, IHpccErrorCode, IHpccErrorType, etc.  The HpccError constructors take these interfaces rather than the enums. This allows you to create a separate set of error codes for wsclient, clienttools, or whatever project this is used in, without having to go back and update the base enum; just create an error code enum like 

class WsclientErrorCode implements IHpccErrorCode

and you're good to go.

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [X] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [X] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [X] I have given due consideration to all of the following potential concerns:
  - [X] Scalability
  - [X] Performance
  - [ ] Security
  - [X] Thread-safety
  - [ ] Premature optimization
  - [X] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

## Testing:

I created two unit tests that exercise the two new classes and demonstrate how they're used.

### CODE REVIEW

**Comment by rpastrana** [comment]
@drealeed

**Comment by rpastrana** [comment]
@drealeed please refactor to new subproject: https://github.com/hpcc-systems/HPCC-JAPIs/tree/master/commons-hpcc

File: hpcc_commons/src/main/java/org/hpccsystems/commons/errors/HpccErrorLevel.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public enum HpccErrorLevel {
+    FATAL,
+    ERROR,
+    WARNING,
+    INFO;
+    
+    public static HpccErrorLevel getErrorLevel(String eclenum) {
+        if (eclenum==null) {
+            return null;
+        } else if ("1".equals(eclenum) || "2".equals(eclenum)) {
```
Comment (rpastrana) [inline hpcc_commons/src/main/java/org/hpccsystems/commons/errors/HpccErrorLevel.java]: can we convert the string to an Integer, and change the if else if to a switch?

**Comment by rpastrana** [comment]
@drealeed this PR needs to be refactored

**Comment by drealeed** [comment]
Closing this for now, will resubmit as a pr in commons.

**Pull request closed without merging**